Intersect.Server/Maps/MapInstance.cs

```diff
@@ -329,60 +331,33 @@ public void DespawnItems()
 
         public void DespawnNpcsOf(NpcBase npcBase)
         {
-            var npcs = new List<Npc>();
-            lock (GetMapLock())
+            foreach (var entity in mEntities.Values)
             {
-                foreach (var entity in mEntities)
+                if (entity.GetType() == typeof(Npc) && ((Npc)entity).Base == npcBase)
                 {
-                    if (entity.GetType() == typeof(Npc) && ((Npc) entity).Base == npcBase)
-                    {
-                        npcs.Add((Npc) entity);
-                    }
-                }
-
-                foreach (var en in npcs)
-                {
-                    en.Die(0);
+                    entity.Die(0);
                 }
             }
         }
 
         public void DespawnResourcesOf(ResourceBase resourceBase)
         {
-            var resources = new List<Resource>();
-            lock (GetMapLock())
+            foreach (var entity in mEntities.Values)
             {
-                foreach (var entity in mEntities)
-                {
-                    if (entity.GetType() == typeof(Resource) && ((Resource) entity).Base == resourceBase)
-                    {
-                        resources.Add((Resource) entity);
-                    }
-                }
-
-                foreach (var en in resources)
+                if (entity.GetType() == typeof(Resource) && ((Resource)entity).Base == resourceBase)
                 {
-                    en.Die(0);
+                    entity.Die(0);
                 }
             }
         }
 
         public void DespawnProjectilesOf(ProjectileBase projectileBase)
         {
-            var projectiles = new List<Projectile>();
-            lock (GetMapLock())
+            foreach (var entity in mEntities.Values)
             {
-                foreach (var entity in mEntities)
-                {
-                    if (entity.GetType() == typeof(Projectile) && ((Projectile) entity).Base == projectileBase)
-                    {
-                        projectiles.Add((Projectile) entity);
-                    }
-                }
-
-                foreach (var en in projectiles)
+                if (entity.GetType() == typeof(Projectile) && ((Projectile)entity).Base == projectileBase)
```

Comment:
This should be `if (entity is Projectile projectile && projectile.Base == projectileBase)`.